### PR TITLE
[BUILD] Upgrading versions

### PR DESCRIPTION
**Pull request by CRamsan**
This PR contains a few changes bundled into one. Changes are: 

- Updated several packages to their latest versions. No breaking changes detected.
- Updated the JS target to use the Kotlin MPP plugin instead of the deprecated Kotlin JS plugin.

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -43,3 +43,7 @@ kotlin.native.useEmbeddableCompilerJar=true
 kotlin.mpp.androidSourceSetLayoutVersion=2
 # Enable kotlin/native experimental memory model
 kotlin.native.binary.memoryModel=experimental
+
+# Gradle will show a warning when a Kotlin/Native target cannot be built on this machine.
+# Disable such message.
+kotlin.native.ignoreDisabledTargets=true
```
Comment (CRamsan) [inline gradle.properties:49]: This is used to remove an annoying warning message while building through gradle.  The message is about unsopported platforms(ie mac/ios) not being able to be build.

File: gradle/kotlin-js-lib.gradle [hunk 2]
```diff
@@ -2,22 +2,31 @@
  * Configure a Kotlin-JS project with safe defaults.
  */
 
-apply plugin: "org.jetbrains.kotlin.js"
+apply plugin: "org.jetbrains.kotlin.multiplatform"
```
Comment (CRamsan) [inline gradle/kotlin-js-lib.gradle:5]: The kotlin.js plugin is being deprecated in favor of the kotlin multiplatform plugin.

File: gradle/kotlin-js-lib.gradle [hunk 3]
```diff
@@ -2,22 +2,31 @@
...
+                implementation "org.jetbrains.kotlinx:kotlinx-coroutines-core-js:_"
+            }
+        }
+        jsTest {
+            dependencies {
+                implementation "org.jetbrains.kotlin:kotlin-test-js:_"
+                implementation "org.jetbrains.kotlinx:kotlinx-coroutines-test:_"
+            }
+        }
+    }
```
Comment (CRamsan) [inline gradle/kotlin-js-lib.gradle:29]: Same as before, but using the Kotlin MP plugin instead.

File: gradle/kotlin-js-lib.gradle [hunk 4]
```diff
@@ -27,7 +36,7 @@ def releaseJs = tasks.register("releaseJs") {
 
     dependsOn 'detekt'
     dependsOn 'compileKotlinJs'
-    dependsOn 'test'
+    dependsOn 'jsTest'
```
Comment (CRamsan) [inline gradle/kotlin-js-lib.gradle:39]: When using the Kotlin MP plugin, the task name is `jsTest`.

File: gradle/kotlin-mpp-compose-lib.gradle [hunk 5]
```diff
@@ -68,7 +68,7 @@ kotlin {
         }
     }
 
-    android() {
+    androidTarget() {
```
Comment (CRamsan) [inline gradle/kotlin-mpp-compose-lib.gradle:71]: In Kotlin 1.9.20, the target name for Android is changed from `android` to `androidTarget`.

**Review by gaaliciA1990 — APPROVED** [review]
Looks good

**Pull request merged**